### DISCUSSION TITLE
БД

### DISCUSSION THREAD

**Discussion by MariaDanu** [DaovaJo/Isys_engineering#4] [Ideas]
Как я вижу БД, но это еще не точно. Пока я вижу четыре таблицы:
1. Пользователи: Код пользователя, Фамилия, Имя. Четвертое поле будет либо телефон, либо +2 поля это Логин и Пароль. Надо точно решить как наш пользователь будет авторизовываться
2. Словарь: Слово, Перевод (или замена), значение (по поводу значения, тоже надо смотреть по ситуации, будет ли у нас доступ к словарю для пользователя, чтобы он посмотрел значения данного слова, либо будем это поле делать отдельной таблицей)
3. Чаты: Код чата, Наименование
4. Сообщение: Доступ, Пользователь (FK табл. Пользователи), Чат (FK табл. чаты), Словарь (FK табл. Словарь)
Если есть свои предложения, то пишите. Будем обсуждать, все таки лучше будет когда думают не одна голова, а несколько.

**Comment by MariaDanu** [comment]
Пока рисунок не стала делать, но если не понятно со слов. Могу нарисовать и прикрепить

**Reply by DaovaJo** [reply]
считаю нужным зафигачить схему а то правда не совсем понятно

**Reply by MariaDanu** [reply]
ахахха, как-будто со схемой тебе станет легче

**Reply by MariaDanu** [reply]
Сейчас сделаю

**Comment by MariaDanu** [comment]
https://github.com/DaovaJo/Isys_engineering/tree/main/%D0%A1%D1%85%D0%B5%D0%BC%D1%8B%20%D0%91%D0%94
Закинула свой вариант структуры БД, смотрите и жду от вас ответы

**Reply by morozenka09** [reply]
Это как мне кажется наиболее просто и незумудрëнный вариант. Я за такую БД.

**Reply by vikulkin** [reply]
я тоже согласна,все по делу,молодец,Маш)

**Reply by MariaDanu** [reply]
Я также жду ваших вариантов, вдруг у кого-то будет лучше структура

**Reply by vikulkin** [reply]
![таблица](https://user-images.githubusercontent.com/96170343/194895417-04dc929a-f44c-4fbe-a415-3c0f62b31032.jpg)
вот мой пример,мне кажется замена и перевод слов нужно добавить в доступ,а в словаре оставить как раз таки только значение слов и сам их список

**Comment by MariaDanu** [comment]
я покидала пару ссылок, вдруг они понадобятся:
https://learn.microsoft.com/ru-ru/skypeforbusiness/schema-reference/persistent-chat-database-schema/sample-persistent-chat-database-queries
https://habr.com/ru/company/tensor/blog/483176/
https://ruseller.com/lessons.php?id=1864&rub=37

**Comment by Bibitriy** [comment]
Хранения таблиц
Таблица пользователя -> таблица чата с другим пользователем
нужно составить новую схему

**Comment by PolMaka2** [comment]
Короче, темы для обсуждения в среду:
1. Есть матерные слова, которые ничего не несут, никакой смысловой нагрузки
2. У каждого пользователей при регистрации должен автоматически появлятся чат с админом, куда он сможет вносить свою интерпретацию тех или иных слов

**Comment by MariaDanu** [comment]
(https://qna.habr.com/q/621828)
